src/pip/_internal/req/constructors.py

```diff
@@ -319,7 +319,7 @@ def install_req_from_req_string(
     try:
         req = Requirement(req_string)
     except InvalidRequirement:
-        raise InstallationError("Invalid requirement: '%s'" % req)
+        raise InstallationError("Invalid requirement: '%s'" % req_string)
```

Comment:
I ran into this bug while making typos writing the unit tests

Reply:
If you had time for it, it would be great if you could do this in a small, separate PR for this with a unit test.

Reply:
Done: https://github.com/pypa/pip/pull/6419 I'll update this PR to remove that fix